File: Running Sum of 1d Array_easy.js
/**
 	* Time Complexity: O(n)
 	* Space Complexity: O(n)
 */
/**
 * @param {number[]} nums
 * @return {number[]}
 */
var runningSum = function(nums) {
    ans = []
    for (let i in nums){
        if (i == 0){
            ans.push(nums[i])
            continue
        }
        ans.push(nums[i] + ans[i - 1])
    }
    return ans
};

File: Running Sum of 1d Array_easy.py
'''
    Time: O(n)
    Space: O(n)
'''
class Solution:
    def runningSum(self, nums: List[int]) -> List[int]:
        ans = []

        for i, elt in enumerate(nums):
            if i == 0:
                ans.append(elt)
                continue
            ans.append(elt + ans[i - 1])
        return ans